tests/test_transformation.py

```diff
@@ -0,0 +1,311 @@
+                        "properties":   {
+                            "type": {"enum": ["residential", "business"]}
+                        }
+                    }
+                ]
+            }
+        }
+    }
+    expected = {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+
+        "definitions": {
+            "address": {
+                "type": "object",
+                "properties": {
+                    "street_address": {"type": "string"}
+                }
+            }
+        },
+
+        "type": "object",
+
+        "properties": {
+            "billing_address": {
+                "type": "object",
+                "properties": {
+                    "street_address": {"type": "differenttype"}
+                }
+            },
+            "shipping_address": {
+                "allOf": [
+                    {
+                        "type": "object",
+                        "properties": {
+                            "street_address": {"type": "string"}
+                        }
+                    },
+                    {
+                        "properties":   {
+                            "type": {"enum": ["residential", "business"]}
+                        }
+                    }
+                ]
+            }
+        }
+    }
+    result = resolve_references(
+        v1,
+        'file://' + os.path.join(
+            os.path.dirname(os.path.realpath(__file__)),
+            'test_outside',
+            'test_outside'
+        )
+    )
+    assert expected == result
+
+
+def test_internal_resolving():
+    """Test internal resolving passes."""
+    v1 = {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+
+        "definitions": {
+            "^[a-zA-Z0-9]+$": {"type": "string"}
+        },
+
+        "type": "object",
+
+        "properties": {
+            "^[a-zA-Z]+$": {"$ref": "#/definitions"}
+        }
+    }
+    expected = {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+
+        "definitions": {
+            "^[a-zA-Z0-9]+$": {"type": "string"}
+        },
+
+        "type": "object",
+
+        "properties": {
+            "^[a-zA-Z]+$": {
+                "^[a-zA-Z0-9]+$": {"type": "string"}
+            }
+        }
+    }
+    result = resolve_references(v1)
+    assert expected == result
+
+
+def test_modify_schema_option():
+    """Test if modify_schema option set to False not modifies schema."""
+    v1 = {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+
+        "definitions": {
+            "address": {
+                "type": "object",
+                "properties": {
+                    "street_address": {"type": "string"}
+                }
+            }
+        },
+
+        "type": "object",
+
+        "properties": {
+            "billing_address": {"$ref": "#/definitions/address"},
+        }
+    }
+    output = resolve_references(v1, modify_schema=False)
+    assert output is not v1
```

Comment:
Can you also check that in this case `v1` was not modified?
And also run with `modify_schema = True` and check that `output` is `v1`.